src/librustc_codegen_llvm/va_arg.rs

```diff
@@ -107,12 +107,30 @@ pub(super) fn emit_va_arg(
     // is lacking in some instances, so we should only use it as a fallback.
     let arch = &bx.cx.tcx.sess.target.target.arch;
     match (&**arch,
-           bx.cx.tcx.sess.target.target.options.is_like_windows) {
-        ("x86", true) => {
+           bx.cx.tcx.sess.target.target.options.is_like_windows,
+           bx.cx.tcx.sess.target.target.options.is_like_osx) {
```

Comment:
This is correct by circumstance, but is not ideal. Technically if there was a `aarch64-apple-darwin` spec it would match here and generate a generic Aarch64 ABI `VaList` structure and then use the `char *` variant of the `va_arg` instruction. Is there a better way to match against IOS here other than using the `target_os` string?

Reply:
CC @rust-lang/compiler ^

Reply:
:+1: updated